### PR TITLE
Use cached grpc ssl credential for ObjC library

### PR DESCRIPTION
**Pull request by muxi**
gRPC ObjC library had been reading the default pem file's contents and feed them into `grpc_ssl_credential_create`, caused multiple apps to crash due to excessive memory usage. This PR changes the usage of `grpc_ssl_credential_create` to use the cached credentials which is more preferable and avoid further crashes.

### CODE REVIEW

File: include/grpc/grpc_security.h [hunk 1]
```diff
@@ -191,6 +191,12 @@ typedef struct {
      try to get the roots set by grpc_override_ssl_default_roots. Eventually,
      if all these fail, it will try to get the roots from a well-known place on
      disk (in the grpc install directory).
+
+     It is a known issue that when the default root certificates are offered
+     with this parameter, occasionally an excessive amount of memory (~10MB) is
```
Comment (jiangtaoli2016) [inline include/grpc/grpc_security.h]: This is inaccurate and misleading.  The root cert store is ~1MB. But gRPC opens multiple subchannels, each subchannel has its own SSL context and root store. 

I would rephrase as

gRPC has implemented root certificates cache if the underlying OpenSSL library supports it. The gRPC root certificates cache is only applicable on the default root certificates. If user provides his or her own pem_root_certs when creating an SSL credential object, gRPC would not be able to cache it.

Comment (muxi) [reply include/grpc/grpc_security.h]: looks good :) I'll rephrase

File: include/grpc/grpc_security.h [hunk 2]
```diff
@@ -191,6 +191,15 @@ typedef struct {
      try to get the roots set by grpc_override_ssl_default_roots. Eventually,
      if all these fail, it will try to get the roots from a well-known place on
      disk (in the grpc install directory).
+
+     gRPC has implemented root cache if the underlying OpenSSL library supports
+     it. The gRPC root certificates cache is only applicable on the default
+     root certificates, which is used when this parameter is nullptr. If user
+     provides their own pem_root_certs, when creating an SSL credential object,
+     gRPC would not be able to cached it, and each subchannel will generate a
```
Comment (dapengzhang0) [inline include/grpc/grpc_security.h]: s/cached/cache/

**Comment by jiangtaoli2016** [comment]
Thanks much @muxi for quick fix!
